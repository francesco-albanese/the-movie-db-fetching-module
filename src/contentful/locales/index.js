import { contentfulClient, urls } from '#contentful/contentful.const'
import { transformLocaleResponse } from '#transformation'
import { parseStringifiedData } from '#utils'

export const getLocales = async() => {
  try {
    const { data } = await contentfulClient.get(urls.space, {
      transformResponse: [
        data => parseStringifiedData(data),
        data => transformLocaleResponse(data)
      ]
    })
    
    return data

  } catch (e) {
    throw e
  }
}