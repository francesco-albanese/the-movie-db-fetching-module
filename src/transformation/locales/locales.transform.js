import { isEmpty } from 'lodash-es'

export const transformLocaleResponse = data => {
  
  if (isEmpty(data)) {
    return []
  }

  const { locales } = data
  
  return locales
}