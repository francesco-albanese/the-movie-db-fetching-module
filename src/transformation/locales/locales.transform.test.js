import { transformLocaleResponse } from './locales.transform'
import * as Fixtures from './__fixtures__/locales.transform.fixtures'

describe('transformLocaleResponse', () => {
  it('returns an empty array if data is missing', () => {
    expect(transformLocaleResponse()).toEqual([])
  })

  it('returns transformed locales', () => {
    const result = transformLocaleResponse(Fixtures.data)
    expect(result).toMatchSnapshot()
  })
})