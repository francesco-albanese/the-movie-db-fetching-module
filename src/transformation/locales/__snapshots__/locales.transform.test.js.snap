// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformLocaleResponse returns transformed locales 1`] = `
Array [
  Object {
    "code": "en-GB",
    "default": true,
    "fallbackCode": null,
    "name": "English (United Kingdom)",
  },
  Object {
    "code": "it",
    "default": false,
    "fallbackCode": "en-GB",
    "name": "Italian",
  },
]
`;
